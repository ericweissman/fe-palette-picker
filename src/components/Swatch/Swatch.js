import React, { Component } from 'react';

class Swatch extends Component {
  constructor() {
    super()
    this.state = {
      locked: false,
    }
  }

  toggleLocked = (e) => {
    this.props.handleClick(e)
    this.setState({
      locked: !this.state.locked
    })
  }

  render() {
    const { palette, value } = this.props
    const { locked } = this.state
    return (
      <div style={{ backgroundColor: palette }} className='color-individual'>
        <div className='lock-control'>
          <button onClick={this.toggleLocked} value={value} className={locked ? 'locked' : 'unlocked'}></button>
          <h4>{palette}</h4>
        </div>
      </div>
    )
  }
}

export default Swatch;