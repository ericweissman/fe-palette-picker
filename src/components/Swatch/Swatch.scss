.locked {
  background: url(./images/locked.svg);
  background-size: cover;
  height: 35px;
  width: 35px;
  border: none;
  cursor: pointer;
  outline: none;
  margin-bottom: 40px;
}
.unlocked {
  background: url(./images/unlocked.svg);
  background-size: cover;
  height: 35px;
  width: 35px;
  border: none;
  cursor: pointer;
  outline: none;
  margin-bottom: 40px;
}

.color-individual {
  height: 175px;
  width: 175px;
  display: flex;
  align-items: flex-end;
  justify-content: center;
}

.lock-control {
  h4 {
    font-size: 1.15rem;
    font-weight: 400;
    color: black;
    text-shadow:
    -.5px -.5px 0 #ffff,
    .5px -.5px 0 #ffff,
    -.5px .5px 0 #ffff,
    .5px .5px 0 #ffff;
  }
  flex-direction: column;
  display: flex;
  margin-bottom: 15px;
  align-items: center;
}

.generator-btns {
  margin: 10px;
  button {
    font-size: 20px;
    padding: 10px 15px;
    margin: 10px;
    border-radius: 20px;
    cursor: pointer;
    outline: none;
  }
}