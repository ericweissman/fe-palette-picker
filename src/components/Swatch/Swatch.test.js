import React from 'react';
import { Swatch } from './Swatch'
import { shallow } from 'enzyme'

describe('Swatch', () => {
  let wrapper
  const mockProps = {
    handleClick: jest.fn(),
    value: '1'
  }
  it('should match the correct snapshot', () => {
    wrapper = shallow(<Swatch {...mockProps}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should have the proper default state', () => {
    const expectedState = {
      locked: false,
    }
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should toggle state when toggleLocked is called', () => {
    const expectedState = {
      locked: true,
    }
    wrapper.instance().toggleLocked()
    expect(wrapper.state()).toEqual(expectedState)
  })
})