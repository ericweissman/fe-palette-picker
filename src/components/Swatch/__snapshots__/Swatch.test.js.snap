// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swatch should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Swatch
    handleClick={[MockFunction]}
    value="1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="lock-control"
      >
        <button
          className="unlocked"
          onClick={[Function]}
          value="1"
        />
        <h4 />
      </div>,
      "className": "color-individual",
      "style": Object {
        "backgroundColor": undefined,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="unlocked"
            onClick={[Function]}
            value="1"
          />,
          <h4 />,
        ],
        "className": "lock-control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "unlocked",
            "onClick": [Function],
            "value": "1",
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="lock-control"
        >
          <button
            className="unlocked"
            onClick={[Function]}
            value="1"
          />
          <h4 />
        </div>,
        "className": "color-individual",
        "style": Object {
          "backgroundColor": undefined,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="unlocked"
              onClick={[Function]}
              value="1"
            />,
            <h4 />,
          ],
          "className": "lock-control",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "unlocked",
              "onClick": [Function],
              "value": "1",
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
